lib/device-connections-factory.js

```diff
@@ -37,29 +37,57 @@ class iProxy {
       }
     });
     const status = new B((resolve, reject) => {
-      this.serverSocket.once('listening', resolve);
-      this.serverSocket.once('error', reject);
+      this.localServer.once('listening', resolve);
+      this.localServer.once('error', reject);
     });
-    this.serverSocket.listen(this.localport);
-    await status;
-    this.serverSocket.on('error', (e) => this.log.warn(e.message));
+    this.localServer.listen(this.localport);
+    try {
+      await status;
+    } catch (e) {
+      this.localServer = null;
+      throw e;
+    }
+    this.localServer.on('error', (e) => this.log.warn(e.message));
+
+    this.onBeforeProcessExit = () => {
+      if (this.localServer?.listening) {
```

Comment:
Make sure that the dependency is set to require at least `appium-gulp-plugins@5.1.0` or this won't build.

Reply:
sure